dapp_stats/dapp_stats.py

```diff
@@ -0,0 +1,52 @@
+from collections import defaultdict
+import pydantic
+from typing import DefaultDict, Dict, Optional
+
+from dapp_manager import DappManager
+
+from .exceptions import DappStatsException
+from .statistics.models import NodeStat
+from .statistics.schemas import StateLogEntry
+
+
+class DappStats:
+    def __init__(self, app_id: str):
+        self._app_id = app_id
+
+    def _iter_app_states(self):
+        return DappManager(self._app_id).storage.iter_file("state")
+
+    def get_stats(self) -> Dict:
+        nodes_stats: DefaultDict[str, Dict[int, NodeStat]] = defaultdict(dict)
+        app_stats: Optional[NodeStat] = None
+
+        for raw_state in self._iter_app_states():
+            try:
+                app_state = StateLogEntry.parse_raw(raw_state)
+            except pydantic.ValidationError:
+                raise DappStatsException(
+                    f"dApp {self._app_id } state log is corrupted. Unable to generate statistics."
+                )
+
+            if app_stats is not None:
```

Comment:
couldn't we just initialize `app_stats` with an "empty` `NodeStat` above and then always use `+=` here?

Reply:
I had this like it before but I didn't like how I need to make  `NodeStatistics` being able to handle empty stat. I picked this solution as  lesser evil.